app/models/workers_scaling/resources_api.rb

```diff
@@ -0,0 +1,160 @@
+  # * workers_count
+  def get_infrastructure_statistics(infrastructure_name, params = {})
+    begin
+      {
+          workers_count: InfrastructureFacadeFactory.get_facade_for(infrastructure_name)
+                             .count_sm_records(user_id, experiment_id)
+                             # TODO this loads all sm records form db
+      }
+    rescue InfrastructureErrors::NoSuchInfrastructureError
+      nil
+    end
+  end
+
+  ##
+  # Returns workers records for all available infrastructures
+  def get_all_workers_records(params = {})
+    Hash[map_available_infrastructures method(:get_workers_records), params]
+  end
+
+  ##
+  # Returns workers records for given infrastructure
+  def get_workers_records_by_infrastructure(infrastructure_name, params = {})
+    # TODO load from db only specific parameters
+    begin
+      InfrastructureFacadeFactory.get_facade_for(infrastructure_name)
+          .get_sm_records(user_id, experiment_id, params) || []
+    rescue InfrastructureErrors::NoSuchInfrastructureError
+      nil
+    end
+  end
+
+  ##
+  # Returns workers records for given ids and infrastructure
+  def get_workers_records_by_ids(ids, infrastructure_name)
+    # TODO load from db only specific parameters
+    ids.map {|id| get_worker_record id, infrastructure_name}
+  end
+
+  ##
+  # Returns worker record for given id and infrastructure
+  def get_worker_record(id, infrastructure_name)
+    # TODO load from db only specific parameters
+    begin
+      InfrastructureFacadeFactory.get_facade_for(infrastructure_name)
+          .get_sm_record_by_id(id)
+    rescue InfrastructureErrors::NoSuchInfrastructureError
+      nil
+    end
+  end
+
+  ##
+  # Returns workers records for all available infrastructures
+  def get_all_workers(params = {})
+    Hash[map_available_infrastructures method(:get_workers), params]
+  end
+
+  ##
+  # Returns worker for given infrastructure
+  def get_workers_by_infrastructure(infrastructure_name, params = {})
+    begin
+      InfrastructureFacadeFactory.get_facade_for(infrastructure_name)
+          .yield_simulation_managers(user_id, experiment_id, params) {|x| x}
+    rescue InfrastructureErrors::NoSuchInfrastructureError
+      nil
+    end
+  end
+
+  #
+  # Returns workers for given ids and infrastructure
+  def get_workers_by_ids(ids, infrastructure_name)
+    ids.map {|id| get_worker id, infrastructure_name }
+  end
+
+  ##
+  # Returns worker for given id and infrastructure
+  def get_worker(id, infrastructure_name)
+    begin
+      InfrastructureFacadeFactory.get_facade_for(infrastructure_name)
+          .yield_simulation_manager(get_worker_record(id, infrastructure_name)) {|x| x}
+    rescue InfrastructureErrors::NoSuchInfrastructureError
+      nil
+    end
+  end
+
+  ##
+  # Executes command on workers with on given infrastructure
+  # Possible commands is in #execute_command_on_worker
+  def execute_command_on_workers(infrastructure_name, command)
```

Comment:
Dodałbym jeszcze execute_command_on_all_workers, niezależnie od infrastruktury. Nie wiem, czy to będzie potrzebne, ale łatwo to będzie dopisać.